adapters/repos/db/lsmkv/commitlogger_parser.go

```diff
@@ -57,65 +73,64 @@ func (p *commitloggerParser) Do() error {
 		}
 
 		if err != nil {
-			return errors.Wrap(err, "read commit type")
+			errUnexpectedLength = errors.Wrap(err, "read commit type")
+			break
 		}
 
 		switch commitType {
+		case CommitTypeCollection:
+			f.Close()
+			return errors.Errorf("found a collection commit on a replace bucket")
 		case CommitTypeReplace:
 			if err := p.parseReplaceNode(); err != nil {
-				return errors.Wrap(err, "read replace node")
-			}
-		case CommitTypeCollection:
-			if err := p.parseCollectionNode(); err != nil {
-				return errors.Wrap(err, "read collection node")
+				errUnexpectedLength = errors.Wrap(err, "read replace node")
+				break
 			}
 		}
 	}
 
-	return f.Close()
-}
-
-func (p *commitloggerParser) parseReplaceNode() error {
-	n, err := ParseReplaceNode(p.reader, p.memtable.secondaryIndices)
-	if err != nil {
-		return err
-	}
+	if p.strategy == StrategyReplace {
```

Comment:
is this check needed? `p.doReplace()` is already called only for `StrategyReplace` in `p.Do()`

Reply:
Great catch, thank you. I originally had `.doReplace()` and `.doCollection()` as a single method, but it got too messy, so I split them out. That's probably where the `if` statement was leftover from. Have removed it.